doc/reference/dune/env.rst

```diff
@@ -51,6 +51,10 @@ Fields supported in ``<settings>`` are:
 - ``(js_of_ocaml (runtest_alias <alias-name>))`` specifies the alias under which
   :ref:`inline_tests` and tests (:ref:`tests-stanza`) run for the `js` mode.
 
+- ``(js_of_ocaml (submodes <submodes>))`` controls whether to generate
+  JavaScript, Wasm code, or both. Each submode is either ``js`` or ``wasm``.
+  The default is to generate JavaScript code.
```

Comment:
Instead of having this, which controls JS vs WASM generation globally, couldn't we add a `wasm` constructor to the `(modes)` field in `executable` stanzas (similar to how there is a `js` constructor)?

Reply:
This directive can also be included in `executable` stanzas.

I'm not completely happy with the design either. It might be better to have a separate `wasm_of_ocaml` field, for instance, to be able to pass different flags to the compilers.

Still, I think we need a global way to specify whether we want to generate JavaScript and/or Wasm code.

Reply:
Maybe we could have these in `env` stanzas:
```
(js_of_ocaml (by_default_if <blang expression>))
(wasm_of_ocaml (by_default_if <blang expression>))
```
and a `(jsoo default)` executable mode.

And the mode `(jsoo js)` and `(jsoo wasm)` could be used to generate explicitly some Wasm or JavaScript code.

Reply:
https://tc39.es/source-map/#section-object doesn't mention it

Reply:
Agreed that this submodes stuff doesn't seem satisfactory. Making it more similar to regular modes would be preferable.